aspnetcore/host-and-deploy/health-checks.md

```diff
@@ -0,0 +1,527 @@
+
+To use the sample app to demonstrate management port configuration, create the *launchSettings.json* file in a *Properties* folder.
+
+The following *launchSettings.json* file isn't included in the sample app's project files and must be created manually.
+
+*Properties/launchSettings.json*:
+
+```json
+{
+  "profiles": {
+    "SampleApp": {
+      "commandName": "Project",
+      "commandLineArgs": "",
+      "launchBrowser": true,
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development",
+        "ASPNETCORE_URLS": "http://localhost:5000/;http://localhost:5001/",
+        "ASPNETCORE_MANAGEMENTPORT": "5001"
+      },
+      "applicationUrl": "http://localhost:5000/"
+    }
+  }
+}
+```
+
+Register health check services with `AddHealthChecks` in `Startup.ConfigureServices`. The call to `UseHealthChecks` specifies the management port (*ManagementPortStartup.cs*):
+
+[!code-csharp[](health-checks/samples/2.x/HealthChecksSample/ManagementPortStartup.cs?name=snippet1&highlight=12,18)]
+
+> [!NOTE]
+> You can avoid creating the *launchSettings.json* file in the sample app by setting the URLs and management port explicitly in code. In *Program.cs* where the `WebHostBuilder` is created, add a call to `UseUrls` and provide the app's normal response endpoint and the management port endpoint. In *ManagementPortStartup.cs* where `UseHealthChecks` is called, specify the management port explicitly.
+>
+> *Program.cs*:
+>
+> ```csharp
+> return new WebHostBuilder()
+>     .UseConfiguration(config)
+>     .UseUrls("http://localhost:5000/;http://localhost:5001/")
+>     .ConfigureLogging(builder =>
+>     {
+>         builder.SetMinimumLevel(LogLevel.Trace);
+>         builder.AddConfiguration(config);
+>         builder.AddConsole();
+>     })
+>     .UseKestrel()
+>     .UseStartup(startupType)
+>     .Build();
+> ```
+>
+> *ManagementPortStartup.cs*:
+>
+> ```csharp
+> app.UseHealthChecks("/health", port: 5001);
+> ```
+
+To run the management port configuration scenario using the sample app, execute the following command from the project's folder in a command shell:
+
+```console
+dotnet run --scenario port
+```
+
+## Health Check Publisher
+
+When an `IHealthCheckPublisher` is added to the service container, the health check system periodically executes your health checks and calls `PublishAsync` with the result. This is useful in a push-based health monitoring system scenario that expects each process to call the monitoring system periodically in order to determine health.
+
+The `IHealthCheckPublisher` interface has a single method:
+
+```csharp
+Task PublishAsync(HealthReport report, CancellationToken cancellationToken);
```

Comment:
I think it's also worth mentioning beatpulse here as they have publishers for a few systems. AI is here - more to come https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/tree/master/src/HealthChecks.Publisher.ApplicationInsights